eodatasets3/prepare/sentinel_sinergise_L1_prepare.py

```diff
@@ -0,0 +1,224 @@
+
+    source_system = xmldoc.getElementsByTagName("System")[0].firstChild.data
+    creator_version = xmldoc.getElementsByTagName("Creator_Version")[0].firstChild.data
+    creation_date = xmldoc.getElementsByTagName("Creation_Date")[0].firstChild.data
+    datastrip_metadata = xmldoc.getElementsByTagName("File_Name")[0].firstChild.data
+
+    return {
+        "source_system": source_system,
+        "creator_version": creator_version,
+        "creation_date": creation_date,
+        "datastrip_metadata": datastrip_metadata,
+    }
+
+
+def prepare_and_write(
+    output_yaml_path: Path,
+    ds_path: Path,
+    product_info: Dict,
+    metadata_xml: Dict,
+    format_correctness: Dict,
+) -> Tuple[uuid.UUID, Path]:
+    with DatasetAssembler(
+        metadata_path=output_yaml_path, dataset_location=ds_path
+    ) as p:
+        p.datetime = product_info["timestamp"]
+        p.properties["eo:instrument"] = "MSI"
+        p.properties["eo:platform"] = "sentinel-2a"
+        p.properties["odc:dataset_version"] = "1.0.0"
+        p.properties["odc:producer"] = "sinergise.com"
+        p.properties["odc:product_family"] = "level1"
+        p.properties["odc:processing_datetime"] = (
+            format_correctness["creation_date"].split("=")[1].replace("T", " ")
+        )
+        p.properties["eo:sun_elevation"] = metadata_xml["solar_zenith"]
+        p.properties["eo:sun_azimuth"] = metadata_xml["solar_azimuth"]
+        p.properties["eo:gsd"] = metadata_xml["resolution"]
+        p.properties["eo:cloud_cover"] = metadata_xml["cloud"]
+        p.properties["sentinel:sinergise_product_name"] = product_info[
+            "synergise_product_name"
+        ]
+        p.properties["sentinel:sinergise_product_id"] = product_info[
+            "synergise_product_id"
+        ]
+        p.properties["odc:file_format"] = "JPEG2000"
+        p.properties["odc:region_code"] = product_info["region_code"]
+        p.properties["sentinel:data_type"] = "Level-1C"
+        p.properties["sentinel:product_type"] = "S2MSI1C"
+        p.properties["sentinel:software_version"] = format_correctness[
+            "creator_version"
+        ]
+        p.properties["sentinel:source_system"] = format_correctness["source_system"]
+        p.properties["sentinel:datastrip_metadata"] = format_correctness[
+            "datastrip_metadata"
+        ]
+        p.properties["sentinel:downlink_priority"] = metadata_xml["downlink_priority"]
+        p.properties["sentinel:datastrip_id"] = metadata_xml["datastrip_id"]
+
+        directory = [f for f in listdir(ds_path) if isfile(join(ds_path, f))]
+
+        for dataset in directory:
+            if ".jp2" in dataset and "preview" not in dataset and "TCI" not in dataset:
+                band = dataset.replace(".jp2", "").replace("B", "")
+                name = SENTINEL_MSI_BAND_ALIASES[band]
+                path = ds_path / dataset
+                p.note_measurement(path=path, name=name)
```

Comment:
It would be nice to use `p.note_accessory()` too, to add the the original `xml` and other files too. 

Ideally every useful file in their package is included: either as a measurement or as an accessory file.

Reply:
done